platform/ios/src/MGLMapView.mm

```diff
@@ -1566,7 +1604,11 @@ - (void)handleDoubleTapGesture:(UITapGestureRecognizer *)doubleTap
             [self animateWithDelay:MGLAnimationDuration animations:^
              {
                  [weakSelf unrotateIfNeededForGesture];
+                 [weakSelf notifyScaleBarGestureDidEnd];
              }];
+        } else {
+            [self unrotateIfNeededForGesture];
+            [self notifyScaleBarGestureDidEnd];
```

Comment:
> Have we considered programmatic zooming?

MapKit only shows the scale in response to user gestures, not in response to programmatic camera changes. However, there is an `-[MKMapCamera showsScale]` property that determines whether the scale bar is always visible: if the developer wants the scale bar to be visible during programmatic zooming, they can set and unset `showsScale` around that change to the map’s region or camera.